app/lib/config/__tests__/config-spec.js

```diff
@@ -0,0 +1,254 @@
+/**
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.
+ *
+ * Camunda licenses this file to you under the MIT; you may not use this file
+ * except in compliance with the MIT License.
+ */
+
+const Config = require('..');
+
+const fs = require('fs');
+const path = require('path');
+
+
+describe('Config', function() {
+
+  describe('default', function() {
+
+    let file;
+
+    beforeEach(function() {
+      file = fs.readFileSync(getAbsolutePath('fixtures/config.json'), { encoding: 'utf8' });
+    });
+
+    afterEach(function() {
+      fs.writeFileSync(getAbsolutePath('fixtures/config.json'), file, { encoding: 'utf8' });
+    });
+
+
+    describe('#get', function() {
+
+      it('should get', function() {
+
+        // given
+        const config = new Config({
+          userPath: getAbsolutePath('fixtures')
+        });
+
+        // when
+        const value = config.get('foo');
+
+        // then
+        expect(value).to.equal(42);
+      });
+
+
+      it('should get nested', function() {
+
+        // given
+        const config = new Config({
+          userPath: getAbsolutePath('fixtures')
+        });
+
+        // when
+        const value = config.get([ 'bar', 'baz' ]);
+
+        // then
+        expect(value).to.equal(true);
+      });
+
+    });
+
+
+    describe('set', function() {
+
+      it('should set', function() {
+
+        // given
+        const config = new Config({
+          userPath: getAbsolutePath('fixtures')
+        });
+
+        // when
+        config.set('foo', false);
+
+        // then
+        const value = config.get('foo');
+
+        expect(value).to.equal(false);
+      });
+
+
+      it('should set nested (multiple keys)', function() {
+
+        // given
```

Comment:
Do we really want / need merge functionality here? Just wondering.

Reply:
That is the question, my friend. 🤔 💭

Reply:
I 💘  React and love me a good old `setState({ merge: 'me' })`.

Reply:
Not a convincing argument to me. Especially the fact that we use `merge` under the hood (which React does not) sounds odd to me.

If I wanted to merge, I could do that from within my plug-in easily:

```javascript
const config = await getConfig('key');

const mergedConfig = mergeAllThings(config, . ....);

await setConfig('key', mergedConfig);
```

Reply:
Alright, I'll remove the merging. What about nested properties via multiple keys? It would allow us to set nested properties without having to get the config first.

Reply:
You could set config like this `set('foo.bar', { VALUEOFFOOBAR });`

Reply:
Not sure about the multiple keys thing.

Reply:
Alright, we'll simplify it to `get(key)` and `set(key)`. 👍

Reply:
> You could set config like this `set('foo.bar', { VALUEOFFOOBAR });`

This is not possible if we allow keys like `bpmn.elementTemplates`.